R/fcast.R

```diff
@@ -191,7 +191,10 @@ dcast.data.table = function(data, formula, fun.aggregate = NULL, sep = "_", ...,
   if (run_agg_funs) {
     fun.call = aggregate_funs(fun.call, lvals, sep, ...)
     maybe_err = function(list.of.columns) {
-      if (any(lengths(list.of.columns) != 1L)) stopf("Aggregating function(s) should take vector inputs and return a single value (length=1). However, function(s) returns length!=1. This value will have to be used to fill any missing combinations, and therefore must be length=1. Either override by setting the 'fill' argument explicitly or modify your function to handle this case appropriately.")
+      if (!all(lengths(list.of.columns) == 1L)) {
+        funf = if (is.null(fill)) stopf else warningf #TODO change to always stopf #6329
+        funf("Aggregating function(s) should take a vector as input and return a single value (length=1), but they do not, so the result is undefined. Please fix by modifying your function so that a single value is always returned.")
```

Comment:
This won't get picked up in our .pot file for translation, but I think that's fine as this will just be deleted in the near future anyway, no need to invest too much in ensuring it's translated.

Reply:
actually i believe it will stay as an error so we should translate

Reply:
msg<-gettext(...)
stop(msg, domain=NA)
warning(msg, domain=NA)